<launch>



    <arg name="parfor" default="true"/>

    <arg name="spinner" default="false"/>

    <param name="enable_parallel_loop" value="$(arg parfor)"/>
    
    <node pkg="sparse_stereo" type="image_blur" name="image_blur">
        <remap from="/image_blur/left/image_raw" to="/camera/left/image_raw" />
        <remap from="/image_blur/left/camera_info" to="/camera/left/camera_info"/>
        <remap from="/image_blur/right/image_raw" to="/camera/right/image_raw"/>
        <remap from="/image_blur/right/camera_info" to="/camera/right/camera_info"/>
    </node>

    <node pkg="sparse_stereo" type="pips_sparse_stereo_node" name="sparse_stereo" output="screen" >
        <!--<remap from="/camera/left/image_raw" to="/multisense_sl/camera/left/image_raw"/>
        <remap from="/camera/left/camera_info" to="/multisense_sl/camera/left/camera_info"/>
        <remap from="/camera/right/image_raw" to="/multisense_sl/camera/right/image_raw"/>
        <remap from="/camera/right/camera_info" to="/multisense_sl/camera/right/camera_info"/>-->
    </node>


    <node name="pips_controller_node" pkg="turtlebot_trajectory_testing" type="depth_controller" output="screen" args="">
        <remap from="/camera/depth/image_raw" to="/sparse_stereo/sparse_depth" />
        <remap from="/camera/depth/camera_info" to="/sparse_stereo/camera_info" />
        <param name="odom_spinner" value="$(arg spinner)"/>
    </node>


  
</launch>
